### PR TITLE
feat: updated minty action to expire token in post processing

### PR DESCRIPTION
**Pull request by bradegler**
Moved the call to the minty server and into a new javascript action called `minty`. I then added a post section that calls the GitHub API to revoke the token in the cleanup phase.

I will follow up with changes to the `mint-token` action to call `minty` once the `minty` action gets on main.

### CODE REVIEW

File: .github/actions/minty/main.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const core = require('@actions/core');
```
Comment (verbanicm) [inline .github/actions/minty/main.js:1]: We should add license, all over

File: .github/actions/minty/post.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const core = require('@actions/github');
+const github = require('@actions/github');
```
Comment (verbanicm) [inline .github/actions/minty/post.js:2]: Same import?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -153,6 +153,15 @@ jobs:
           service_url: '${{ env.INTEGRATION_SERVICE_URL }}'
           requested_permissions: '{"scope":"integ","repositories":["github-token-minter"],"permissions":{"issues":"read"}}'
 
+      - id: 'mint-github-token-2'
```
Comment (verbanicm) [inline .github/workflows/ci.yml:156]: Does the existing action ned to be updated? I think the inputs are different right?

File: .github/actions/minty/main.js (hunk 1 above)
Comment (bradegler) [reply .github/actions/minty/main.js:1]: Yes, that's part of the final cleanup. Do you feel that this implementation is correct? Mostly just looking for early input on if this is directionally the "right" way to do this.

File: .github/actions/minty/post.js (hunk 2 above)
Comment (bradegler) [reply .github/actions/minty/post.js:2]: `main.js` and `post.js` are loaded at separate times. They both need imports for the things they need.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (bradegler) [reply .github/workflows/ci.yml:156]: Yes, this was just a test to make sure that it works conceptually. I'll need to land the `minty` action on main, then update the original action to look like `mint-token-2`.

File: .github/actions/minty/post.js (hunk 2 above)
Comment (verbanicm) [reply .github/actions/minty/post.js:2]: No I mean you are importing the same thing twice. I think you want `const core = require('@actions/core')` instead.

Comment (bradegler) [reply .github/actions/minty/post.js:2]: oh geez, yeah, you are right

File: .github/actions/minty/main.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    // get the response as text so we can try to parse JSON
+    const responseText = await response.text();
+
+    if (response.ok) {
+      try {
+        // try to parse response as JSON for older versions of github-token-minter
+        // expecting `{ "token": "TOKEN" }` format
+
+        const resp = JSON.parse(responseText);
```
Comment (verbanicm) [inline .github/actions/minty/main.js:44]: I would use `core.setSecret(resp.token)` prior to setting the output on this response value so its masked in logs.

File: .github/actions/minty/main.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        core.setOutput('token', resp.token);
+      } catch (err) {
+        // we didnt get a JSON response, response body contains the token
+        // so just use the response text
+
+        const token = responseText;
```
Comment (verbanicm) [inline .github/actions/minty/main.js:50]: Same here.

File: .github/actions/minty/main.js (hunk 1 above)
Comment (verbanicm) [reply .github/actions/minty/main.js:17]: Overall process LGTM.

File: .github/actions/minty/main.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        core.setOutput('token', token);
+        core.exportVariable('minty-token', token);
```
Comment (verbanicm) [inline .github/actions/minty/main.js:52]: nit: `MINTY_TOKEN`

File: .github/actions/minty/main.js (hunk 1 above)
Comment (bradegler) [reply .github/actions/minty/main.js:17]: 👍

File: .github/actions/minty/dist-main/index.js
Comment (verbanicm) [inline .github/actions/minty/dist-main/index.js:1]: nit: I think we use dist/main/index.js in our google github actions.

**Review by verbanicm — APPROVED** [review]

File: .github/actions/minty/package.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "main.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build-min": "ncc build -o dist/main -m main.js && ncc build -o dist/post -m post.js",
+    "build": "ncc build -o dist/main main.js && ncc build -o dist/post post.js"
+  },
+  "author": "",
+  "license": "ISC",
```
[the remark this answers is not in the document]
Comment (bradegler) [reply .github/actions/minty/package.json]: fixed

**Pull request merged**